src/webportal/src/app/job/job-view/fabric/job-detail/components/summary.jsx

```diff
@@ -96,82 +118,84 @@ export default class Summary extends React.Component {
       return;
     }
 
+    const HintItem = ({header, value}) => (
+      <div>
+        <span className={c(t.w4, t.dib)} style={{fontWeight: FontWeights.semibold}}>
+          {header}
+        </span>
+        <span>{value}</span>
+      </div>
+    );
+
     const state = getHumanizedJobStateString(jobInfo);
     if (state === 'Failed') {
-      const diag = jobInfo.jobStatus.appExitDiagnostics;
-      const code = jobInfo.jobStatus.appExitCode;
-      if (code === 177) {
-        // user code error
-        let userExitCode;
-        if (diag) {
-          let match = diag.match(/<Raw>\[ExitCode\]: (\d+)/);
-          if (match) {
-            userExitCode = parseInt(match[1], 10);
-          }
+      const result = [];
+      const spec = jobInfo.jobStatus.appExitSpec;
+      // error type
+      const type = spec && spec.type;
+      if (type) {
+        result.push(<HintItem key='type' header='Error Type:' value={spec.type} />);
+      }
+      // trigger task
+      const role = jobInfo.jobStatus.appExitTriggerTaskRoleName;
+      const idx = jobInfo.jobStatus.appExitTriggerTaskRoleIndex;
+      if (role) {
```

Comment:
Can we show all the spec fields on web?

- code: 196
  phrase: CONTAINER_OOM_KILLED_BY_DOCKER
  issuer: PAI_RUNTIME
  causer: USER_CONTAINER
  type: USER_FAILURE
  stage: RUNNING
  behavior: NON_TRANSIENT
  reaction: NEVER_RETRY
  reason: "Container killed by docker due to it exceeded the request memory"
  repro:
    - "User program uses more memory than its requested"
  solution:
    - "Increase per task memory request"
    - "Decrease per task memory usage by such as increasing task number"

Reply:
Below fields can skipped if you want, others pls keep.
issuer
causer
stage
behavior



---
In reply to: [277197923](https://github.com/Microsoft/pai/pull/2550#discussion_r277197923) [](ancestors = 277197923)

Reply:
@scarlett2018 to decide.